roadmap.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "Apache Flink Roadmap"
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<hr />
+
+{% toc %}
+
+**Preamble:** This is not an authoritative roadmap in the sense of a strict plan with a specific
+timeline. Rather, we, the community, share our vision for the future and give an overview of the bigger
+initiatives that are going on and are receiving attention. This roadmap shall give users and
+contributors an understanding where the project is going and what they can expect to come.
+
+The roadmap is continuously updated. New features and efforts should added to the roadmap once
+there is consensus that they will happen and what they will roughly look like for the user.
+
+# Analytics, Applications, an the roles of DataStream, DataSet, and Table API
+
+Flink views stream processing as a [unifying paradigm for data processing]({{ site.baseurl }}/flink-architecture.html)
+(batch and real-time) and event-driven applications. The APIs are evolving to reflect that view:
+
+  - The **Table API / SQL** is becoming the primary API for analytical use cases, in a unified way
+    across batch and streaming. To support analytical use cases in a more streamlined fashion,
+    the API is [extended with additional functions](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739)
```

Comment:
`the API is [extended with additional functions]` -> `the API  enhancement [FLIP-29](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=97552739) that extended with additional functions.`